src/main/java/org/opentripplanner/apis/gtfs/GraphQLScalars.java

```diff
@@ -50,7 +55,57 @@ public String parseLiteral(Object input) {
     )
     .build();
 
-  public static GraphQLScalarType geoJsonScalar = GraphQLScalarType
+  public static final GraphQLScalarType offsetDateTimeScalar = GraphQLScalarType
+    .newScalar()
+    .name("OffsetDateTime")
+    .coercing(
+      new Coercing<OffsetDateTime, String>() {
+        @Override
+        public String serialize(@Nonnull Object dataFetcherResult)
+          throws CoercingSerializeException {
+          if (dataFetcherResult instanceof ZonedDateTime zdt) {
+            return zdt.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+          } else if (dataFetcherResult instanceof OffsetDateTime odt) {
+            return odt.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+          } else {
+            throw new CoercingSerializeException(
+              "Cannot serialize object of class %s".formatted(
+                  dataFetcherResult.getClass().getSimpleName()
+                )
+            );
+          }
+        }
+
+        @Override
+        public OffsetDateTime parseValue(Object input) throws CoercingParseValueException {
+          if (input instanceof CharSequence cs) {
+            return OffsetDateTimeParser.parseLeniently(cs).orElseThrow(() -> valueException(input));
+          }
+          throw valueException(input);
+        }
+
+        @Override
+        public OffsetDateTime parseLiteral(Object input) throws CoercingParseLiteralException {
+          if (input instanceof StringValue sv) {
+            return OffsetDateTimeParser
+              .parseLeniently(sv.getValue())
+              .orElseThrow(() -> literalException(input));
+          }
+          throw literalException(input);
+        }
+
+        private static CoercingParseValueException valueException(Object input) {
+          return new CoercingParseValueException("Cannot parse %s".formatted(input));
+        }
+
+        private static CoercingParseLiteralException literalException(Object input) {
+          return new CoercingParseLiteralException("Cannot parse %s".formatted(input));
+        }
+      }
```

Comment:
I'm not sure if these are necessary. At least when I tested other scalar, it automatically outputted the value that was used when an exception was thrown.

Reply:
Or to be more precise, the input doesn't need to be inserted into the exception.

Reply:
Would be probably be more helpful to state something like that the value was not a valid offset datetime.

Reply:
I'm sorry, actually only the literal logs the input value by default, `parseValue` does not when exception is thrown so you should include the input value in that case.